src/Find_Minimum_in_Rotated_Sorted_Array.java

```diff
@@ -0,0 +1,67 @@
+// Time Complexity : O(log n)
+// Space Complexity : O(1)
+// Did this code successfully run on Leetcode : Yes
+// Any problem you faced while coding this : No
+
+
+// Your code here along with comments explaining your approach
+/* if mid is less than mid-1, then second half is sorted in ascending order and return mid as minimum
+ * if mid is greater than next element, then second half is sorted in ascending order and return mid+1 as minimum */
+
+class Find_Minimum_in_Rotated_Sorted_Array {
+
+	public static int findMin(int[] nums) {
+		if(nums == null || nums.length == 0)
+			return -1;
+		
+		int low = 0;
+        int high = nums.length -1;
+        
+        if(nums[high] > nums[0] || nums.length == 1)	//if length of array is 1, then return first element
+          return nums[0];
+        
+        while(low <= high) {
+        	int mid = low + (high - low) /2;
+        	
+        	if(nums[mid] < nums[mid-1])			// if mid is less than mid-1, then second half is sorted in ascending order and return mid as minimum
+        		return nums[mid];
+        	if(nums[mid] > nums[mid+1])			// if mid is greater than next element, then second half is sorted in ascending order and return mid+1 as minimum
```

Comment:
Please perform bounds check, what if mid is 0 and you do mid -1 or mid is last element and you perform mid+1. This will crash for some cases.